lib/blacklight/access_controls/search_builder.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+module Blacklight
+  module AccessControls
+    # SearchBuilder that restricts access via Solr.
+    #
+    # Note: solr_access_filters_logic is an Array of Symbols.
+    # It sets defaults. Each symbol identifies a _method_ that must be in
+    # this class, taking two parameters (permission_types, ability).
+    # Can be changed in local apps or by plugins, e.g.:
+    #   SearchBuilderWithAccessControls.solr_access_filters_logic += [:new_method]
+    #   SearchBuilderWithAccessControls.solr_access_filters_logic.delete(:we_dont_want)
```

Comment:
Shouldn't this be Blacklight::AccessControls::SearchBuilder instead of SearchBuilderWithAccessControls?  Same for the line above it.